components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```diff
@@ -151,11 +153,12 @@ public AccessTokenInfo getTokens(String appName, String requestURL, String grant
                 // Access token for authorization code grant type
                 APIMAppConfigurations appConfigs = ServiceReferenceHolder.getInstance()
                         .getAPIMAppConfiguration();
-                String callBackURL = appConfigs.getApimBaseUrl() + AuthenticatorConstants.AUTHORIZATION_CODE_CALLBACK_URL + appName;
+                String callBackURL = appConfigs.getApimBaseUrl() + AuthenticatorConstants.AUTHORIZATION_CODE_CALLBACK_URL + appName
+                        + "?uiService=" + uiServiceURL;
                 // Get the Authorization Code
                 if (requestURL.contains("code=")) {
                     String requestURLQueryParameters = requestURL.split("\\?")[1];
-                    String authorizationCode = requestURLQueryParameters.split("=")[1].split("&")[0];
+                    String authorizationCode = requestURLQueryParameters.split("&code=")[1].split("&")[0];
```

Comment:
as we discussed we need to get rid of splitting logic.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/4843/commits/36e8f7d5e5ad289b70f148ce5df8914e56eb85aa by adding the query parameter for the `authentication code`.